### PR TITLE
Cam C

### PR DESCRIPTION
**Pull request by ckccameron**

### CODE REVIEW

**Comment by ckccameron** [comment]
Add Week 1 Independent Challenge work

File: test/school_test.rb [hunk 1]
```diff
@@ -45,11 +41,12 @@ def test_it_can_add_student_names
...
-    assert_equal '16:00', school1.end_time
-    assert_equal '12:00', school2.end_time
+    assert_equal '16:00', school.end_time
   end
-end
+
+  def test_it_is_full_time
+    school = School.new("9:00", 7)
+
+    assert_equal true, school.is_full_time?
```
Comment (KathleenScriver) [inline test/school_test.rb:52]: There are 2 `end` missing to this test which is causing syntax errors and causing your code to error out. You'll want to get into the habit of running your test suite after each test you write and after each method you write to ensure there are no syntax errors in your code.

File: lib/school.rb [hunk 2]
```diff
@@ -1 +1,47 @@
...
+  end
+
+  def student_names
+  p @student_names
+
+  def is_full_time
+end
+end
+
+school = School.new ("9:00", 7)
```
Comment (KathleenScriver) [inline lib/school.rb:35]: Looks like you were testing here, but we want to move away from testing like this and instead use the test suite to test your class and all the methods in the class.

File: lib/school.rb [hunk 3]
```diff
@@ -1 +1,47 @@
+class School
 
+  attr_reader :start_time,
+              :hours_in_school_day,
+              :student_names
+
+  def initialize (start_time, hours_in_school_day)
```
Comment (KathleenScriver) [inline lib/school.rb:7]: Want to have your parenthesis touching the method name, no space.

File: lib/school.rb [hunk 4]
```diff
@@ -1 +1,47 @@
...
+    @start_time = start_time
+    @hours_in_school_day = hours_in_school_day
+    @student_names = []
+  end
+
+  def start_time (start_time)
```
Comment (KathleenScriver) [inline lib/school.rb:13]: Here, you are defining a method called `start_time`. Do you already have a method named `start_time`? Hint: what is the attr_reader on line 3 doing?

File: lib/school.rb [hunk 5]
```diff
@@ -1 +1,47 @@
...
+    @start_time = start_time
+  end
+
+  def hours_in_school_day (hours)
```
Comment (KathleenScriver) [inline lib/school.rb:17]: Be careful here, the interaction pattern and tests did not ask for this method to take an argument.

File: lib/school.rb [hunk 6]
```diff
@@ -1 +1,47 @@
...
+    @hours_in_school_day = hours
+  end
+
+  def end_time
+    end_time = @start_time + @hours_in_school_day
+
+  def add_student_name (name)
+    @student_names << name
```
Comment (KathleenScriver) [inline lib/school.rb:25]: Yes, great job adding the argument into the array of student_names

File: test/school_test.rb [hunk 7]
```diff
@@ -45,11 +41,12 @@ def test_it_can_add_student_names
   end
 
   def test_it_can_calculate_end_time
-    skip
     school1 = School.new('9:00', 7)
-    school2 = School.new('9:00', 3)
 
-    assert_equal '16:00', school1.end_time
-    assert_equal '12:00', school2.end_time
```
Comment (KathleenScriver) [inline test/school_test.rb:53]: Careful, you don't want to delete existing tests. They are here to guide your development and help you see the pattern the method is expecting.

File: test/school_test.rb [hunk 8]
```diff
@@ -45,11 +41,12 @@ def test_it_can_add_student_names
...
+    assert_equal '16:00', school.end_time
   end
-end
+
+  def test_it_is_full_time
```
Comment (KathleenScriver) [inline test/school_test.rb:49]: Good title for your test

File: test/school_test.rb (hunk 1 above)
Comment (KathleenScriver) [inline test/school_test.rb:52]: Good setting up the test, you have a new school instance, and a good assertion.

**Pull request closed without merging**